### PR TITLE
RAMLLintBear: Add it

### PR DESCRIPTION
**Pull request by Adrianzatreanu**
Adds RAMLLintBear which ensures consistency and
quality of code.

Fixes https://github.com/coala-analyzer/coala-bears/issues/452

### CODE REVIEW

**Comment by Adrianzatreanu** [comment]
Hm, for some reason the regex fails I think, it doesnt catch
```
[adi@fedora23 yml]$ ramllint test.raml 

error POST http://example.com/resource
  RAML section - method - must include: description [method_desc]

error POST http://example.com/resource
  RAML section - method - must include: examples [request_example]

error POST http://example.com/resource
  RAML section - method - must include: schemas [request_schema]
```

and I don't understand why.

File: bears/yml/RAMLLintBear.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    """
+    RAML Linter is a static analysis, linter-like, utility that will enforce
+    rules on a given RAML document, ensuring consistency and quality.
+    """
+
+    LANGUAGES = "RAML"
+
+    @staticmethod
+    def create_arguments(filename, file, config_file):
+        return filename
```
Comment (SanketDG) [inline bears/yml/RAMLLintBear.py]: the return from `create_arguments` should be a tuple.

`return filename,`

File: bears/yml/RAMLLintBear.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
+from coalib.bearlib.abstractions.Linter import linter
+
+
+@linter(executable='ramllint',
+        use_stdout=False,
+        use_stderr=True,
+        output_format='regex',
+        output_regex=r'(?P<severity>error) (?P<filename>.+)\n  (?P<message>.+)')
```
Comment (arafsheikh) [inline bears/yml/RAMLLintBear.py]: You don't have to catch the filename, and you aren't catching the name of the rule (origin). 
The regex should look something like `(?P<severity>error).*\n   (?P<message>.+) \[(?P<origin>.+)\]` assuming `POST http://example.com/resource` isn't a part of the message.

Comment (arafsheikh) [inline bears/yml/RAMLLintBear.py]: According to https://github.com/QuickenLoans/ramllint/blob/master/src/cli.js#L15 there are probably three logging levels. Modify regex accordingly.

File: bears/yml/RAMLLintBear.py [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+class RAMLLintBear:
+    """
+    RAML Linter is a static analysis, linter-like, utility that will enforce
+    rules on a given RAML document, ensuring consistency and quality.
+    """
+
+    LANGUAGES = "RAML"
+
+    @staticmethod
+    def create_arguments(filename, file, config_file):
```
Comment (arafsheikh) [inline bears/yml/RAMLLintBear.py]: Does the linter take in arguments?

Comment (Adrianzatreanu) [inline bears/yml/RAMLLintBear.py]: not from what I can see

File: bears/yml/RAMLLintBear.py (hunk 2 above)
Comment (Adrianzatreanu) [inline bears/yml/RAMLLintBear.py]: yes, you're right, it takes 3, initially I had them but because it didnt catch any regex I tried with only one as i knew for sure that the file im testing has only errors :D thanks

**Comment by Adrianzatreanu** [comment]
https://asciinema.org/a/a3ofm0pdiei06b6jv6hnwpkg7 ^^ for what it's worth

**Comment by Adrianzatreanu** [comment]
@Makman2 would love your help here. the bear works locally as you can see on the asciinema, but tests fail :/

**Comment by Adrianzatreanu** [comment]
@justuswilhelm this works perfectly locally, wont pass tests though, no idea why

**Comment by Adrianzatreanu** [comment]
@Makman2 as asked, i rebased this and pushed, using the command manually on the same file that holds ``invalid__file`` from tests it works perfectly, but the tests dont pass, no idea why.

File: .ci/deps.sh [hunk 4]
```diff
@@ -29,6 +29,7 @@ sudo dpkg -i hlint_1.9.26-1_amd64.deb
 # NPM commands
 sudo rm -rf /opt/alex # Delete ghc-alex as it clashes with npm deps
 npm install
+npm install -g ramllint
```
Comment (SanketDG) [inline .ci/deps.sh]: this should be moved to `pacakge.json`

File: .ci/deps.sh [hunk 5]
```diff
@@ -57,7 +58,7 @@ done
 
 pip install -r docs-requirements.txt
 
-# Downloading nltk data that's required for nltk to run 
+# Downloading nltk data that's required for nltk to run
```
Comment (SanketDG) [inline .ci/deps.sh]: put these in a separate commit.

**Comment by SanketDG** [comment]
@Adrianzatreanu tests pass now? :smile:

File: bears/yml/RAMLLintBear.py [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+
+@linter(executable='ramllint',
+        output_format='regex',
+        output_regex=r'(?P<severity>error|warning|info).*\n  (?P<message>.+) '
+                     r'\[(?P<origin>.+)\]')
+class RAMLLintBear:
+    """
+    RAML Linter is a static analysis, linter-like, utility that will enforce
+    rules on a given RAML document, ensuring consistency and quality.
+    """
```
Comment (Makman2) [inline bears/yml/RAMLLintBear.py]: like discussed on gitter, please put a note that files should not have leading empty newlines!

File: bears/yml/RAMLLintBear.py [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+@linter(executable='ramllint',
+        output_format='regex',
+        output_regex=r'(?P<severity>error|warning|info).*\n  (?P<message>.+) '
+                     r'\[(?P<origin>.+)\]')
+class RAMLLintBear:
+    """
+    RAML Linter is a static analysis, linter-like, utility that will enforce
+    rules on a given RAML document, ensuring consistency and quality.
+    Note: Files should not have leading empty lines, else the bear fails to
+    identify the problems correctly.
```
Comment (sils1297) [inline bears/yml/RAMLLintBear.py]: can't you check that automatically and remove them making a patch? Nobody reads notes...

Comment (Adrianzatreanu) [inline bears/yml/RAMLLintBear.py]: how should I do that :( ?

Comment (sils1297) [inline bears/yml/RAMLLintBear.py]: ask @Makman2 what to override or read the linter decorator? :P

Comment (Adrianzatreanu) [inline bears/yml/RAMLLintBear.py]: those 2 sound like really good plans! :D

Comment (Makman2) [inline bears/yml/RAMLLintBear.py]: @sils1297 let's merge that first, and open an issue. In fact it's the fault of ramllint, not ours.

**Comment by sils1297** [comment]
ack https://github.com/coala-analyzer/coala-bears/pull/456/commits/ce446c3b7084976f254ef30869accbe8cb91113a

**Comment by sils1297** [comment]
@rultor merge
